README.rst

```diff
@@ -74,11 +68,140 @@ To install in a virtual environment in your current project:
 
 
 
-Usage Example
-=============
+Usage Examples
+==============
+
+See `examples/ble_file_transfer_simpletest.py` for a client example. A stub server implementation is in `examples/ble_file_transfer_simpletest.py`.
+
+Protocol
+=========
+
+The file transfer protocol is meant to be simple and easy to implement. It uses free space counts as a way to rate limit file content data transfer. All multi-byte numbers are encoded with the least significant byte first ("<" in CPython's struct module).
+
+GATT Service
+--------------
+
+The UUID of the service is `0xfebb`, Adafruit's 16-bit service UUID.
+
+The base UUID used in characteristics is `ADAFxxxx-4669-6C65-5472-616E73666572`. The 16-bit numbers below are substituted into the `xxxx` portion.
+
+The service has two characteristics:
+
+* version (`0x0100`) - Simple unsigned 32-bit integer version number. Always 1.
+* raw transfer (`0x0200`) - Bidirectional link with a custom protocol. The client does WRITE_NO_RESPONSE to the characteristic and then server replies via NOTIFY. (This is similar to the Nordic UART Service but on a single characteristic rather than two.) The commands over the transfer characteristic are idepotent and stateless. A disconnect during a command will reset the state.
```

Comment:
```suggestion
* raw transfer (`0x0200`) - Bidirectional link with a custom protocol. The client does WRITE_NO_RESPONSE to the characteristic and then server replies via NOTIFY. (This is similar to the Nordic UART Service but on a single characteristic rather than two.) The commands over the transfer characteristic are idempotent and stateless. A disconnect during a command will reset the state.
```